### PR TITLE
add network foreach cryprto in getPrices

### PR DESCRIPTION
**Pull request by raniamorheg**
**Dear team,**

**This PR** introduces an important update to the cryptoDetails variable by adding the networkSupported and contractAddress fields for each cryptocurrency. The information for these fields will be retrieved using the following API endpoint: https://pro-api.coinmarketcap.com/v2/cryptocurrency/info.

The following changes have been made:

**Adding networkSupported and contractAddress Fields**: We have enhanced the cryptoDetails variable by including the networkSupported and contractAddress fields for each cryptocurrency. These fields provide information about the supported network and contract address associated with each cryptocurrency. The data for these fields will be fetched from the specified API endpoint.
Reviewers, please ensure that the networkSupported and contractAddress fields have been successfully added to the cryptoDetails variable by fetching the necessary data from the provided API endpoint. Verify that the information is accurate and aligned with the respective cryptocurrencies. Your feedback, suggestions, and alternative approaches to improve the implementation of these additions are highly appreciated.

Thank you for your time and effort in enhancing the functionality and data representation of our platform.

Best regards,
Rania Morheg

### CODE REVIEW

File: web3/wallets.js [hunk 1]
```diff
@@ -505,6 +506,51 @@ exports.getAllWallets = async (req, res) => {
...
+            },
+            headers: {
+                'X-CMC_PRO_API_KEY': process.env.CMCAPIKEY,
+            },
+           
+        }
+       
+        var results 
+        try {
+            results = await Promise.all([ rp.request(options)])
```
Comment (hichri-louay) [inline web3/wallets.js]: As I understand, Promise.all is generally used when dealing with multiple promises. However, it seems like we're only making a single request in this instance. If we're only making a single request, we can call this request directly without using Promise.all.

File: web3/wallets.js [hunk 2]
```diff
@@ -505,6 +506,51 @@ exports.getAllWallets = async (req, res) => {
...
+
+        } catch (error) {
+            throw new Error('Error fetching  networks')
+        }
+   let networksContract = []
+     const  result = await results[0].data.data
```
Comment (hichri-louay) [inline web3/wallets.js]: Consider removing await in this line (539). As I can see from the previous recap of the data at line 533, it seems unnecessary here. The Promise.all is already handling the promise resolution.

File: web3/wallets.js [hunk 3]
```diff
@@ -624,12 +679,30 @@ exports.getPrices = async () => {
...
+          
+           
+         }
+    }catch(error){
+        throw new Error('Error fetching networks')
+    }
+
+
 
             for (var i = 0; i < token200.length; i++) {
```
Comment (hichri-louay) [inline web3/wallets.js:692]: It seems like there are quite a few loop methods within the getPrice function, which might be causing a performance hit, especially when dealing with large data sets. Could you provide more context about the function and the data it's handling? With more information, we can discuss potential ways to refactor the function to make it more efficient. Specifically, we might look at alternatives to the loop methods being used and determine whether we can minimize their usage, or consider different data structures that could provide the same functionality but with better performance.

File: web3/wallets.js [hunk 4]
```diff
@@ -624,12 +679,30 @@ exports.getPrices = async () => {
...
+            
+              idcrypto.push(priceMap[i].id.toString())
                 finalMap[priceMap[i].symbol] = priceMap[i]
+        
                 delete finalMap[priceMap[i].symbol].symbol
             }
+            
+    try{
+        networksContract= await   getNetworkByToken( idcrypto.join(','))
+        for (var i = 0; i < idcrypto?.length; i++) {
```
Comment (hichri-louay) [inline web3/wallets.js]: It seems like there are quite a few loop methods within the getPrice function, which might be causing a performance hit, especially when dealing with large data sets. Could you provide more context about the function and the data it's handling? With more information, we can discuss potential ways to refactor the function to make it more efficient. Specifically, we might look at alternatives to the loop methods being used and determine whether we can minimize their usage, or consider different data structures that could provide the same functionality but with better performance.

File: web3/wallets.js [hunk 5]
```diff
@@ -624,12 +679,30 @@ exports.getPrices = async () => {
         }
 
             var finalMap = {}
+            var idcrypto =[]
             for (var i = 0; i < priceMap.length; i++) {
```
Comment (hichri-louay) [inline web3/wallets.js]: It seems like there are quite a few loop methods within the getPrice function, which might be causing a performance hit, especially when dealing with large data sets. Could you provide more context about the function and the data it's handling? With more information, we can discuss potential ways to refactor the function to make it more efficient. Specifically, we might look at alternatives to the loop methods being used and determine whether we can minimize their usage, or consider different data structures that could provide the same functionality but with better performance.

File: web3/wallets.js [hunk 6]
```diff
@@ -559,8 +606,14 @@ exports.getPrices = async () => {
...
             response.data.data.push(responseSattJet.data.data.BTT)
             let  priceMap
+            let networksContract
             try {
-             priceMap = response.data.data.map((elem) => {
+             
+    
+                
+               
+             priceMap = response.data.data.map( (elem) => {
```
Comment (hichri-louay) [inline web3/wallets.js:605]: It seems like there are quite a few loop methods within the getPrice function, which might be causing a performance hit, especially when dealing with large data sets. Could you provide more context about the function and the data it's handling? With more information, we can discuss potential ways to refactor the function to make it more efficient. Specifically, we might look at alternatives to the loop methods being used and determine whether we can minimize their usage, or consider different data structures that could provide the same functionality but with better performance.

**Review by hichri-louay — CHANGES_REQUESTED** [review]

File: web3/wallets.js [hunk 7]
```diff
@@ -505,6 +506,51 @@ exports.getAllWallets = async (req, res) => {
...
+    
+        const options = {
+            method: 'GET',
+            url: process.env.CMC_CRYPTO_DETAILS,
+            params: {
+                id: idCrypto
+            },
+            headers: {
+                'X-CMC_PRO_API_KEY': process.env.CMCAPIKEY,
+            },
```
Comment (skanderKha) [inline web3/wallets.js]: I think it's better to  move the options variable in the config file to make cointMarketCap reusiable for futur needs .

**Review by skanderKha — APPROVED** [review]

**Review by HamdiBenK — APPROVED** [review]

**Pull request merged**